fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java

```diff
@@ -66,7 +66,16 @@ public List<String> buildCollectSQLList() {
             for (String columnName : columns) {
                 sqlInUnion.add(buildCollectFullStatisticSQL(db, table, partition, columnName));
                 scanRowCount += partition.getRowCount();
-                if (scanRowCount >= Config.statistic_collect_max_row_count_per_query) {
+                if (scanRowCount >= Config.statistic_collect_max_row_count_per_query
+                        /*
+                         * Because the previous code has filtered empty partitions, the partition row count of 0 here means.
+                         * 1. The load has just been completed, and the number of lines has not been collected.
+                         * 2. Just restarted, the number of lines has not been collected
```

Comment:
For this two case, I think we needn't collect statistic automatically.

Reply:
The current Delete is also a LOAD, so there is a situation where the actual number of rows is 0, because all data is deleted. If a partition with a row count of 0 is completely ignored, the partition may never be updated.